src/libdecaf/src/gpu/opengl/glsl2_alu_reduction.cpp

```diff
@@ -218,22 +218,16 @@ CUBE(State &state, const ControlFlowInst &cf, const std::array<AluInst, 4> &grou
    insertLineEnd(state);
 
    // If any instructions write to a register, copy appropriately from PVo
-   char srcMaskBuf[5], dstMaskBuf[5];
-   int maskPos = 0;
    for (auto i = 0u; i < group.size(); ++i) {
       if (group[i].op2.WRITE_MASK()) {
-         srcMaskBuf[maskPos] = "xyzw"[i];
-         dstMaskBuf[maskPos] = "xyzw"[group[i].word1.DST_CHAN()];
-         maskPos++;
+         char srcMask = "xyzw"[i];
+         char dstMask = "xyzw"[group[i].word1.DST_CHAN()];
+         int dstIndex = group[i].word1.DST_GPR();
+         insertLineStart(state);
+         state.out << "R[" << dstIndex << "]." << dstMask << " = PVo." << srcMask << ";";
```

Comment:
I think this should be using the writeAfterGroup thing that I added the other day.